plugins/org.eclipse.elk.alg.packing.rectangles/src/org/eclipse/elk/alg/packing/rectangles/firstiteration/FirstIteration.java

```diff
@@ -0,0 +1,341 @@
+package org.eclipse.elk.alg.packing.rectangles.firstiteration;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.elk.alg.packing.rectangles.util.DrawingData;
+import org.eclipse.elk.alg.packing.rectangles.util.PackingStrategy;
+import org.eclipse.elk.alg.packing.rectangles.util.DrawingDataDescriptor;
+import org.eclipse.elk.graph.ElkNode;
+
+/**
+ * Class that handles the first iteration of the algorithm, producing an approximated bounding box for the packing.
+ * 
+ * @author dalu
+ */
+public class FirstIteration {
+    /** Desired aspect ratio. */
+    private double dar;
+    /** Packing Strategy. */
+    private PackingStrategy packingStrat;
+    /** Shift when placing behind or below the last placed rectangle. */
+    private boolean lpShift;
+
+    /**
+     * Constructs an object that the first iteration can be executed on.
+     * 
+     * @param desiredAr
+     *            desired aspect ratio.
+     * @param packStrat
+     *            given packing strategy.
+     * @param lpShifting
+     *            indicating whether a shift should happen after placing near the last placed node.
+     */
+    public FirstIteration(final double desiredAr, final PackingStrategy packStrat, final boolean lpShifting) {
+        this.dar = desiredAr;
+        this.packingStrat = packStrat;
+        this.lpShift = lpShifting;
+    }
+
+    /**
+     * Calculates a drawing for the given rectangles according options set in the object creation of
+     * {@link FirstIteration}. This method also sets the coordinates for the rectangles.
+     * 
+     * @param rectangles
+     *            the rectangles to place.
+     * @return a drawing calculated by this methods algorithm.
+     */
+    public DrawingData approxBoundingBox(final List<ElkNode> rectangles) {
+        // init
+        List<ElkNode> placedRects = new ArrayList<>();
+        ElkNode lastPlaced = null;
+        ElkNode toPlace = null;
+        DrawingData opt1, opt2, opt3, opt4, currentValues, bestOpt;
+
+        // place first box.
+        ElkNode firstRect = rectangles.get(0);
+        firstRect.setX(0);
+        firstRect.setY(0);
+        placedRects.add(firstRect);
+        lastPlaced = firstRect;
+        currentValues = new DrawingData(this.dar, firstRect.getWidth(), firstRect.getHeight(),
+                DrawingDataDescriptor.WHOLE_DRAWING);
+
+        // place the other boxes.
+        for (int rectangleIdx = 1; rectangleIdx < rectangles.size(); rectangleIdx++) {
+            toPlace = rectangles.get(rectangleIdx);
+
+            // determine drawing metrics for different candiate positions/placement options
+            DrawingDataDescriptor lpRight = DrawingDataDescriptor.CANDIDATE_POSITION_LAST_PLACED_RIGHT;
+            opt1 = calcValuesForOpt(lpRight, toPlace, lastPlaced, currentValues, placedRects);
+
+            DrawingDataDescriptor lpBelow = DrawingDataDescriptor.CANDIDATE_POSITION_LAST_PLACED_BELOW;
```

Comment:
Don't create variables for the constants, use them directly.

Reply:
If the constants turn out to be too long, consider a `static import`.